index.js

```diff
@@ -23,13 +23,21 @@ class RPCClient {
       }),
       credentials: this.sendCredentials,
     })
-      .then(response => response.json())
-      .then((response) => {
+      .then(({ json, status }) =>
+        new Promise((resolve, reject) => {
+          json()
+            .then(parsedResponse => resolve({
+              response: parsedResponse,
+              status,
+            }))
+            .catch(error => reject(error));
+        }))
+      .then(({ response, status }) => {
         if (response.error) {
           // throw a handled exception
           const err = new Error(response.error);
           err.code = response.code;
-          err.handled = true;
+          err.handled = status === 400;
```

Comment:
Would it make sense to use `error.code` to set the `handled` property here? 

There is a [related comment about this on the `buffer-rpc`](https://github.com/bufferapp/buffer-rpc/pull/4#discussion_r223325726) pull request, if we set a specific code in our error handling middleware, we could rely on it here.

My main concern with using `status === 400` is that the [`createError` helper function](https://github.com/bufferapp/buffer-rpc/blob/master/src/createError.js#L1), allows for `statusCode` to be passed as an option but it is not clear (on call site) that using that parameter would make the error _unhandled_ on the client.

Apart from that, if we rely on `response.code` the implementation would be simpler without requiring the status to be passed along to the next `then()`. 

How does this alternative approach feel to you @hharnisc?